### PR TITLE
minify zone.js, reflect.js for angular2.min.js bundle

### PR DESCRIPTION
**Pull request by dariajung**
#1382

@rkirov 

A little smaller than before.

```
410377 Jun  8 17:14 angular2.min.js
```

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -785,7 +785,7 @@ gulp.task('bundle.js.prod.deps', ['bundle.js.prod'], function() {
 
 gulp.task('bundle.js.min.deps', ['bundle.js.min'], function() {
   return bundler.modify(
-      ['node_modules/zone.js/dist/zone-microtask.js', 'dist/build/angular2.min.js'],
+      ['node_modules/zone.js/dist/zone-microtask.min.js', 'dist/build/angular2.min.js'],
```
Comment (rkirov) [inline gulpfile.js]: Actually, I think bundle might be completely broken as it doesn't have `node_modules/reflect-metadata/Reflect.js` shim that dev bundle has. Can you try to add it minified too?

Here is an example of the hello world dev bundle - http://plnkr.co/edit/tpl:BrUl7z1vvZBGWBfSGS8D

Comment (rkirov) [inline gulpfile.js]: Also remove a TODO from line 721

Comment (dariajung) [inline gulpfile.js]: Is there a way to call minify/uglify with `bundler`? Or should I add gulp-uglify as a dependency and pipe uglify to a task for minifying `reflect.js`?

**Comment by dariajung** [comment]
@rkirov Added gulp-uglify to minify `Reflect.js`. I made the minification its own task, but I don't know if that is the recommended way of going about this.

**Comment by dariajung** [comment]
@googlebot I signed it (the CLA)

**Comment by dariajung** [comment]
@rkirov I think this is ready for code-review. I made the `reflect.js` minification its own gulp task but I don't know if it makes sense to just add `pipe(uglify())` to the `'bundle.js.min.deps'` task.

**Comment by rkirov** [comment]
I would prefer having one less gulp task, and rolling it into bundle.js.min.deps.

The question you should be asking is "would anyone run `gulp bundle.metadata.minify` directly?" and my guess is the answer is no.

**Comment by dariajung** [comment]
@rkirov Ok, uglifying happening in `bundle.js.min.deps` and commits have been squashed to one.

**Comment by jeffbcross** [comment]
I restarted the failing Travis job because it was a flake with the infinite scroll benchmark.

File: package.json [hunk 2]
```diff
@@ -31,6 +31,7 @@
     "gulp-insert": "^0.4.0",
     "gulp-modify": "0.0.5",
     "gulp-replace": "^0.5.3",
+    "gulp-uglify": "^1.2.0",
```
Comment (vicb) [inline package.json]: Why twice ?

Comment (rkirov) [inline package.json]: remove from deps just keep in dev deps.

File: gulpfile.js [hunk 3]
```diff
@@ -778,16 +778,18 @@ gulp.task('bundle.js.sfx.dev', ['build.js.dev'], function() {
 
 gulp.task('bundle.js.prod.deps', ['bundle.js.prod'], function() {
   return bundler.modify(
-      ['node_modules/zone.js/dist/zone-microtask.js', 'dist/build/angular2.js'],
+      ['node_modules/zone.js/dist/zone-microtask.js', 'node_modules/reflect-metadata/Reflect.js', 'dist/build/angular2.js'],
```
Comment (rkirov) [inline gulpfile.js]: this is getting longer than 100 chars, use two lines.

**Comment by dariajung** [comment]
@rkirov

**Comment by rkirov** [comment]
merged.

**Pull request closed without merging**